test/examples/make.jl

```diff
@@ -157,3 +157,35 @@ examples_html_doc = makedocs(
     ],
     doctest = false,
 )
+info("Building mock package docs: HTMLWriter with pretty URLs and canonical links")
+examples_html_doc = makedocs(
+    debug = true,
+    root  = examples_root,
+    build = "builds/html-pretty-urls",
```

Comment:
The build should go into a separate directory. But I think it would actually be fine to just enable `html_canonical` for the normal "pretty URLs" build and not have a separate build. The `html_canonical = nothing` case will be tested by the other HTML build.